hazelcast/src/main/java/com/hazelcast/datalink/DataLinkService.java

```diff
@@ -20,34 +20,89 @@
 import com.hazelcast.core.HazelcastException;
 import com.hazelcast.spi.annotation.Beta;
 
+import java.util.Map;
+
 /**
- * Service for accessing data link factories
+ * Service for managing {@link DataLink}s and their lifecycle.
```

Comment:
Since com.hazelcast.datalink isn't a private package, I'm assuming it is an API.

Is this interface going to be exposed to the clients or is this members-only API? 

Certain elements in this interface are not client friendly like:
```
Datalink getDataLink(String name);
```
Is some kind of remote proxy going to be returned? I'm not crazy about adding more API to clients btw; they are already obese.

If this is a member-only API, then it should be properly documented.

Reply:
If anything this is certainly a member-only API.

I am not really certain how this should be done - most of the methods are for internal use, and the only useful method for users would be `getDataLink` - for users implementing a connector.

Reply:
Reduced the API to getDataLink only
a4f8d95b
dbc14ce5f1f